tests/e2e_tests/subcommands/root/test_root_register_add_member_senate.py

```diff
@@ -13,8 +13,25 @@
 from ...utils import setup_wallet
 
 
+def assert_sequence(lines, sequence):
+    sequence_ptr = 0
+    for line in lines:
+        lineset = set(line.split())
+        seqset = sequence[sequence_ptr]
+        if lineset.intersection(seqset) == seqset:
+            sequence_ptr += 1
+            if sequence_ptr >= len(sequence):
+                # all items seen
+                break
+    assert sequence_ptr == len(
+        sequence
+    ), f"Did not find sequence[{sequence_ptr}] = '{sequence[sequence_ptr]}' in output"
+
+
```

Comment:
Let's maybe make this a bit more readable:

```py
def assert_sequence(lines: list[str], sequence: Collection[set[str]]):
    sequence_ptr = 0
    for line in lines:
        words_in_line = set(line.split())
        current_seq_set = sequence[sequence_ptr]
        if current_seq_set.issubset(words_in_line):
            sequence_ptr += 1
            if sequence_ptr == len(sequence):
                break

    assert sequence_ptr == len(sequence), (
        f"Did not find sequence[{sequence_ptr}] = '{sequence[sequence_ptr]}' in output"
    )
```

Reply:
Other than replacing `>=` with `==`, I agree. I have seen such constructs, where a value has to land on an exact value to break out of a loop, lead to very hard to find bugs.

Reply:
agreed, pushed this